### PR TITLE
Refactor BookmarkAnimeCommand to BookmarkParser to reduce coupling

### LINKED ISSUE #134: Refactor BookmarkAnimeCommand to include BookmarkParser

### PR DESCRIPTION
**Pull request by n3wsoldier**
Add

1. BookmarkParser
2. BookmarkParserTest

Closes #134

### CODE REVIEW

**Review by ChanJianHao — APPROVED** [review]
LGTM 👍🏻

File: src/main/java/seedu/duke/parser/BookmarkParser.java [hunk 1]
```diff
@@ -0,0 +1,126 @@
...
+            String invalidParameter = "Parameter : " + paramParts[0] + " requires an additional field";
+            LOGGER.log(Level.WARNING, "Could not load bookmark command:" + invalidParameter);
+            throw new AniException(invalidParameter);
+        } else if (paramParts.length > 2) {
+            String invalidParameter = "Parameter : " + paramParts[0] + " has too much fields";
+            LOGGER.log(Level.WARNING, "Could not load bookmark command:" + invalidParameter);
+            throw new AniException(invalidParameter);
+        }
+    }
+}
```
Comment (EyoWeiChin) [inline src/main/java/seedu/duke/parser/BookmarkParser.java]: Would this be better if it was refactored using the utility methods available in CommandParser?

Comment (n3wsoldier) [reply src/main/java/seedu/duke/parser/BookmarkParser.java]: Yea I saw, but I was considering keeping my custom messages and logging. 
I think I will take up your advice to keep things consistent.

Comment (n3wsoldier) [reply src/main/java/seedu/duke/parser/BookmarkParser.java]: Thanks for the feedback.

**Review by EyoWeiChin — APPROVED** [review]
Thanks for refactoring Bookmark into the CommandParser!

LGTM!

**Pull request merged**